Add semver support when using pushing git tag

roy-bongers opened the pull request:
I love this GitHub Action but I was missing semver support where Docker images would be pushed with multiple version tags. When pushing a git tag `v1.2.3` this change will push images with the `1.2.3`, `1.2` and `1` tags to the Docker registry. The v prefix is optional and wil NOT be pushed.

elgohr commented:
Hey, thank you for the PR!
We had this discussion in https://github.com/elgohr/Publish-Docker-Github-Action/issues/67

roy-bongers commented:
Just took a look at the suggestion you made [here](https://github.com/elgohr/Publish-Docker-Github-Action/issues/67#issuecomment-586576736). The action suggested force pushes major / minor Git tags to GitHub. I don't like this solution for a couple of reasons:

 - The action is (force) pushing to git which I think is unwanted
 - You'll end up with an additional v1 and v1.2 tag in your git repository
 - The action is pushing to git which should then triggers additional Actions (yours). So in total three actions will run instead of just one
 - It doesn't strip the version prefix `v` which seems the Docker way of tagging

elgohr commented:
Could you please add tests?

roy-bongers commented:
I've added tests and moved the `tag_semver` check above `tag_names` so `tag_names` could act as a default when pushing a tag that doesn't validate the regex.

Currently a tag containing a string suffix is not supported (`-alpha`, `-rc`). It could be caught with `tag_names` to prevent it from being pushed as `latest`. I've thought about adding the suffix to all tags but I don't think that's a good idea. When pushing `1.2.3-rc` you would then also push `1.2-rc` and `1-rc`. But 1.2 and 1 would already be released so it wouldn't make sense.

elgohr commented:
Looks good for now. I think that I'll have the time for the merge in two weeks. Just wanted to tell you in advance
